src/input/devices.rs

```diff
@@ -70,17 +108,38 @@ fn get_controller_pose(
     Some(vr::space_relation_to_openvr_pose(location, velocity))
 }
 
+fn get_generic_tracker_pose(
+    xr_data: &OpenXrData<impl crate::openxr_data::Compositor>,
+    session_data: &SessionData,
+    tracker: &TrackedDevice,
+    origin: vr::ETrackingUniverseOrigin,
+) -> Option<vr::TrackedDevicePose_t> {
+    let space = match tracker.get_type() {
+        TrackedDeviceType::GenericTracker { space, .. } => Some(space),
+        _ => return None,
+    };
+
+    let (location, velocity) = space?
+        .relate(
+            session_data.get_space_for_origin(origin),
+            xr_data.display_time.get(),
+        )
+        .ok()?;
+
+    Some(vr::space_relation_to_openvr_pose(location, velocity))
+}
+
 impl TrackedDevice {
     pub(super) fn new(
         device_type: TrackedDeviceType,
         profile_path: Option<xr::Path>,
         interaction_profile: Option<&'static dyn InteractionProfile>,
     ) -> Self {
         Self {
-            device_type,
             interaction_profile,
             profile_path: profile_path.unwrap_or(xr::Path::NULL),
             connected: device_type == TrackedDeviceType::Hmd,
```

Comment:
```suggestion
            connected: matches!(device_type, TrackedDeviceType::Hmd),
```
Then the `PartialEq` impl for `TrackedDeviceType` can be removed.